### ISSUE TITLE
Question about 1d distillation

### ISSUE THREAD

**Issue by kdwonn** [ShivamDuggal4/adaptive-length-tokenizer#4]
Hi! Thanks for the code and interesting work!

While looking into the code, I'm a bit confused about how the 2D tokens (that would be distilled to 1D) and masked 2D tokens are constructed:

1. According to this line: [link](https://github.com/ShivamDuggal4/adaptive-length-tokenizer/blob/8c0b4ec29c76963ad81ffdd91c9afd901f5215f5/adaptive_tokenizers/adaptive_vqgan.py#L196), it seems like these 2D tokens are not solely constructed from the pre-trained VQGAN, but also by directly patch embedding the original image. Did this design outperform solely using 2D tokens from the pre-trained VQGAN?

2. I'm also confused about way of constructing 2D mask tokens for decoder. It seems like [this implementation](https://github.com/ShivamDuggal4/adaptive-length-tokenizer/blob/8c0b4ec29c76963ad81ffdd91c9afd901f5215f5/adaptive_tokenizers/adaptive_vqgan.py#L106) uses two learnable parameters: `decoder_class_embedding` and `decoder_mask_token`. It seems like an unnecessary component at first glance (using either one would be enough). Is there any reason for using two different learnable parameters here?

**Comment by ShivamDuggal4** [comment]
Hi @kdwonn 

Thank you for your interest in our work! You’ve raised two great points:

1. While we didn’t explicitly ablate the performance differences between using only VQGAN, only learned patch embeddings, or their combination (as in the current version), we opted for the latter for two main reasons: 
    - The latent-distillation encoder’s input features (1024 in the current version) are no longer constrained by VQGAN’s feature dimension (256).
    - The learned patch embedding can serve as an additional source of pixel information that the VQGAN encoder might have overlooked. This is further supported by our second-stage fine-tuning, where we update everything—including the VQGAN encoder and decoder.

2. The `decoder_mask_token` might be redundant given the presence of `decoder_positional_encoding`, though we did not explicitly ablate this. Similarly, class tokens in both the encoder and decoder could also be redundant.

Best Regards,
Shivam
